### PR TITLE
updated Digitizer framework adding (a) option to allow independent El…

### PR DESCRIPTION
**Pull request by suchandradutta**
The SiPhase2 Digitizer package has been updated. The details of the update can be found in following the presentation at the Tracker Phase2 Simulation Meeting 

https://indico.cern.ch/event/536859/contributions/2354905/attachments/1362994/2063384/DigiStudy_Status_Oct28_2016.pdf

@delaere @boudoul @emiglior @atricomi

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
@suchandradutta , in SimTracker/SiPhase2Digitizer/test/DigiTest_cfg.py  hardcoded file in /store/... is proposed instead of local file, which we are trying to avoid - this test soon will not work any more.

Another minor comment: "Digitizer framework" in the title for me is ambiguous, because for CMS framework means something different.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimTracker/SiPhase2Digitizer/test/DigiTest_cfg.py [hunk 1]
```diff
@@ -64,15 +62,20 @@
...
-
-# Automatic addition of the customisation function from SLHCUpgradeSimulations.Configuration.combinedCustoms                                                 
-#from SLHCUpgradeSimulations.Configuration.combinedCustoms import cust_2023tilted
+# customisation of the process.
 
-#call to customisation function cust_2023tilted imported from SLHCUpgradeSimulations.Configuration.combinedCustoms                                          
-#process = cust_2023tilted(process)
 # Automatic addition of the customisation function from SLHCUpgradeSimulations.Configuration.combinedCustoms
-from SLHCUpgradeSimulations.Configuration.combinedCustoms import cust_2023LReco 
+from SLHCUpgradeSimulations.Configuration.combinedCustoms import cust_2023tilted4021
```
Comment (davidlange6) [inline SimTracker/SiPhase2Digitizer/test/DigiTest_cfg.py]: this function doesn't exist at this moment- is it coming or is it now removed thanks to the eras migration (which is where this needs to go at some point)

File: SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc [hunk 2]
```diff
@@ -699,7 +704,7 @@ void Phase2TrackerDigitizerAlgorithm::pixel_inefficiency(const SubdetEfficiencie
 
   // setup the chip indices conversion
   unsigned int Subid=DetId(detID).subdetId();
-  if (Subid == PixelSubdetector::PixelBarrel) { // barrel layers
+  if (Subid == PixelSubdetector::PixelBarrel || Subid == StripSubdetector::TOB) { // barrel layers
```
Comment (suchandradutta) [inline SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc]: @davidlange6  - here TOB means the phase2 barrel layers as defined in the package description of Geometry/TrackerNumberingBuilder in the section of 

"Phase 2 Upgrade Detector DetId schema Tilted Geometry with new pixel detector for TDR (from Version V4021)"

https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Geometry/TrackerNumberingBuilder/README.md

File: SimTracker/SiPhase2Digitizer/test/Phase2TrackerValidateDigi.cc [hunk 3]
```diff
@@ -264,6 +266,7 @@ void Phase2TrackerValidateDigi::bookHistograms(DQMStore::IBooker & ibooker,
 
   std::string top_folder = config_.getParameter<std::string>("TopFolderName");
   std::stringstream folder_name;
+  TH1::SetDefaultSumw2();
```
Comment (dmitrijus) [inline SimTracker/SiPhase2Digitizer/test/Phase2TrackerValidateDigi.cc]: This sets the default behaviour for the rest of the DQM.
Please don't do this.

Comment (suchandradutta) [inline SimTracker/SiPhase2Digitizer/test/Phase2TrackerValidateDigi.cc]: Okay I shall change it

File: SimTracker/SiPhase2Digitizer/test/DigiTest_cfg.py (hunk 1 above)
Comment (suchandradutta) [inline SimTracker/SiPhase2Digitizer/test/DigiTest_cfg.py]: added era option

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**